src/librustc_attr/builtin.rs

```diff
@@ -623,28 +623,22 @@ pub fn eval_condition(
     match cfg.kind {
         ast::MetaItemKind::List(ref mis) if cfg.name_or_empty() == sym::version => {
             try_gate_cfg(cfg, sess, features);
-            let (min_version, span) = match &mis[..] {
-                [NestedMetaItem::Literal(Lit { kind: LitKind::Str(sym, ..), span, .. })] => {
-                    (sym, span)
-                }
-                [NestedMetaItem::Literal(Lit { span, .. })
-                | NestedMetaItem::MetaItem(MetaItem { span, .. })] => {
-                    sess.span_diagnostic
-                        .struct_span_err(*span, &*format!("expected string literal"))
-                        .emit();
-                    return false;
-                }
-                [..] => {
+            let span = match &mis[..] {
+                [] => {
                     sess.span_diagnostic
-                        .struct_span_err(cfg.span, "expected single string literal")
+                        .struct_span_err(cfg.span, "expected a version string")
```

Comment:
I believe it may be better to use `expected a version literal`.

Reply:
It does seem better, changed it.